x-pack/legacy/plugins/epm/public/screens/detail/installation_button.tsx

```diff
@@ -18,23 +23,33 @@ interface InstallationButtonProps {
 export function InstallationButton(props: InstallationButtonProps) {
   const { assets, name, title, version } = props.package;
   const installPackage = useInstallPackage();
+  const { toDetailView } = useLinks();
   const deletePackage = useDeletePackage();
   const getPackageInstallStatus = useGetPackageInstallStatus();
   const installationStatus = getPackageInstallStatus(name);
 
   const isInstalling = installationStatus === InstallStatus.installing;
   const isRemoving = installationStatus === InstallStatus.uninstalling;
   const isInstalled = installationStatus === InstallStatus.installed;
-
   const [isModalVisible, setModalVisible] = useState<boolean>(false);
   const toggleModal = useCallback(() => {
     setModalVisible(!isModalVisible);
   }, [isModalVisible]);
 
+  const installSuccessCallback = useCallback(() => {
+    const packageUrl = toDetailView({ name, version });
+    const dataSourcesUrl = toDetailView({
+      name,
+      version,
+      panel: 'data-sources',
+    });
+    if (window.location.href.includes(packageUrl)) window.location.href = dataSourcesUrl;
```

Comment:
👋 I do plan to merge #52445 in ~1h

Reply:
What i'd like to do is be notified when a request has succeeded and then render a `<Redirect />`.  Similar to this first example: https://tylermcginnis.com/react-router-programmatically-navigate/ .  Having a callback and doing this creates a memory leak error if I navigate away, though because InstallButton is no longer mounted. I could check if it's mounted, but that's a little dirty, so I'm thinking to so spend some time later having the use_package_install hook return the state of the request and the ability to unsubscribe.  This would also remove the need to have to check the URL as it would only redirect of the component was still mounted.

Reply:
@neptunian I am not clear on the notification aspect, but am fine coming back to it. We'll have RR hooks whenever we do.

What I do want to do now is avoid the reload of assigning to location.href and instead update location.hash. Mind doing that in an upcoming PR?

Reply:
@jfsiii I did actually do that in this PR: https://github.com/elastic/kibana/pull/52771/commits/1b61e27d586f28e408459d923609bbea81705aae